BringYourOwnBoard.md

```diff
@@ -0,0 +1,126 @@
+1. The prepared image MUST contain at least three partitions: the A partition, the B partition, and the ADU partition
+2. The prepared image MUST overload or modify the existing bootloader such that the system is able to load into either the A or B partition. The boot logic will be determined by your BSP. You must add the ability to set flags in order to use the correct partition after an update. You can see the one the proof-of-concept agent uses [here](./recipes-bsp/rpi-u-boot-scr/rpi-u-boot-scr.bbappend). Likely you will be able to append the image you're targeting's default bootloader but that will be on a case-by-case basis. 
+3. The image build system MUST bake a hardware compatability file into the image. The Device Update proof-of-concept creates this within the [recipes-support/adu-swupdate-hw-compat.bb](./recipes-support/adu-swupdate-hw-compat/adu-swupdate-hw-compat.bb) but you can choose to use whatever file or versioning method you would like. 
+4. If you are using the [meta-azure-device-update repository](https://github.com/Azure/meta-azure-device-update) meta layer to bake the image in you must set the `ADU_SOFTWARE_VERSION` to the version of Device Update you are using within your build system. 
+
+These are general guidance on larger scale items that need ot be taken care of for the agent to build properly on your device. The following section will first focus on what the Device Update `meta-raspberrypi-adu` layer is attempting to accomplish in each layer and what considerations you may want to make. If you read the above items and are unsure of how to proceed please reach out to the Device Update team or talk to your BSP provider. Likely an answer to your question will require information from both. 
+
+# On Patches In this Repo
+
+There are a lot of patches within this repository. This is a normal process of developing meta-layers for Yocto. They're intended only for the `meta-raspberrypi` layer. You don't need to think about them, however you might find that you do need to update and or edit your existing BSP with some patches of your own. You can safely disregard them in your port. 
+
+# Recipes
+
+## recipes-bsp 
+
+### What does it do? 
+
+Within our project `recipes-bsp` is used for appending instructions to the bootloader that allow SwUpdate to set which partition to boot into after completing an image update (eg if we've completed an update while running on `/dev/mmcblk0p0` then when the device restarts we boot into `/dev/mmcblk0p1` which contains the new version of the image). For our purposes we can use [rpi-u-boot-scr.bbappend](./recipes-bsp/rpi-u-boot-scr/rpi-u-boot-scr.bbappend) to add the [boot.cmd.in](./recipes-bsp/rpi-u-boot-scr/files/boot.cmd.in) file to the Raspberry Pi 4's boot process which instructs the device to do the A/B switching. 
+
+Within this layer we also edit u-boot to use our own firmware environment configuration and append a special patch for a gcc issue that RaspberryPi has been dealing with. 
```

Comment:
Just curious: earlier you indicate that this doc is for getting ADU working on something other than a Raspberry Pi 4. So I was surprised to see the line about a patch for what appears to be an RPi-specific issue. I am not sure what I am supposed to take away from that part?

Reply:
The idea is to explain what our layer is doing and then describe what the customer should do in their port. 

I have no idea what they're going to try and use and what they may try and leave out. I just want to provide them with what we do and recommendations and considerations for what to change. 

I got to this part and went back to add the "On Patches In This Repo" section to explicitly call out they don't need those for that reason. 

I still want to explain why we have those files in there but the reader should be able to interpret that they do not need the patches from the document as a whole.